Sources/Build/BuildOperationBuildSystemDelegateHandler.swift

```diff
@@ -55,6 +55,27 @@ class CustomLLBuildCommand: SPMLLBuild.ExternalCommand {
     ) -> Bool {
         fatalError("subclass responsibility")
     }
+
+    func execute(
+        _ command: SPMLLBuild.Command,
+        _ commandInterface: SPMLLBuild.BuildSystemCommandInterface
+    ) -> BuildValue {
+        let result: Bool = self.execute(command, commandInterface)
+        // FIXME: having to construct a dummy output info here seems problematic.
+        return result ? BuildValue.SuccessfulCommand(outputInfos: [.init(device: 0, inode: 0, mode: 0, size: 0, modTime: .init(seconds: 0, nanoseconds: 0))]) : BuildValue.FailedCommand()
```

Comment:
This needs to get figure out. I suspect that this may be the underlying issue of why the test isn't passing yet.

Reply:
This does not actually seem to be related to the test failure.

Reply:
Ah, I think the issue here is that the output isn't actually match what is happening. If I hack in additional output into `commandStarted`, I do see "C.Foo-debug.module" being started as expected.

Reply:
I do see that we also re-run the compiler if the Swift version doesn't change, though. 

So really two things need to get resolved here:
- we need a way to instrument `swift build` enough to actually test what we need to test
- we need to ensure that the compiler does *not* re-run if the Swift version did not change.

Reply:
I solved this instead by going through the generic always-out-of-date machinery, I didn't realize that was accessible to me. That solves the FIXME as well as the functional issue. Instrumenting didn't turn out to be necessary, verbose output ended up being good enough for that.